nixpkgs_review/review.py

```diff
@@ -423,29 +423,46 @@ def build_pr(self, pr_number: int) -> dict[System, list[Attr]]:
             assert all(system in PLATFORMS for system in self.systems)
             print("-> Fetching eval results from GitHub actions")
 
-            packages_per_system = self.github_client.get_github_action_eval_result(pr)
-            if packages_per_system is None:
-                timeout_s: int = 10
-                print(f"...Results are not (yet) available. Retrying in {timeout_s}s")
-                waiting_time_s: int = 0
-                while packages_per_system is None:
-                    waiting_time_s += timeout_s
-                    print(".", end="")
-                    sys.stdout.flush()
-                    time.sleep(timeout_s)
-                    packages_per_system = (
-                        self.github_client.get_github_action_eval_result(pr)
+            try:
+                packages_per_system = self.github_client.get_github_action_eval_result(
+                    pr
+                )
+                if packages_per_system is None:
+                    timeout_s: int = 10
+                    print(
+                        f"...Results are not (yet) available. Retrying in {timeout_s}s"
+                    )
+                    waiting_time_s: int = 0
+                    while packages_per_system is None:
+                        waiting_time_s += timeout_s
+                        print(".", end="")
+                        sys.stdout.flush()
+                        time.sleep(timeout_s)
+                        try:
+                            packages_per_system = (
+                                self.github_client.get_github_action_eval_result(pr)
+                            )
+                        except ArtifactExpiredError as e:
+                            warn(f"\n{e}. Falling back to local evaluation.")
```

Comment:
Does this respect --eval github and not fallback to local then?

Reply:
I wouldn't want eval to run locally on my laptop as that most likely will OOM kill.